pkg/shim/util_unix.go

```diff
@@ -76,12 +76,16 @@ func AdjustOOMScore(pid int) error {
 const socketRoot = defaults.DefaultStateDir
 
 // SocketAddress returns a socket address
-func SocketAddress(ctx context.Context, socketPath, id string) (string, error) {
+func SocketAddress(ctx context.Context, socketPath, id string, debug bool) (string, error) {
 	ns, err := namespaces.NamespaceRequired(ctx)
 	if err != nil {
 		return "", err
 	}
-	d := sha256.Sum256([]byte(filepath.Join(socketPath, ns, id)))
+	path := filepath.Join(socketPath, ns, id)
+	if debug {
+		path = filepath.Join(path, "debug")
```

Comment:
is is required to add `debug` here? If I understand it correctly, the pprof is using different socket address.